File: win-prog/ProgWinCS/Images and Bitmaps/ImageFromFile/ImageFromFile.cs
//--------------------------------------------
// ImageFromFile.cs © 2001 by Charles Petzold
//--------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class ImageFromFile: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new ImageFromFile());
     }
     public ImageFromFile()
     {
          Text = "Image From File";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Image image = Image.FromFile("..\\..\\..\\Apollo11FullColor.jpg");

          grfx.DrawImage(image, 0, 0);
     }
}

File: win-prog/ProgWinVB/Images and Bitmaps/ImageFromFile/ImageFromFile.vb
'----------------------------------------------
' ImageFromFile.vb (c) 2002 by Charles Petzold
'----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class ImageFromFile
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(new ImageFromFile())
    End Sub

    Sub New()
        Text = "Image From File"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim img As Image = Image.FromFile("..\..\Apollo11FullColor.jpg")
        grfx.DrawImage(img, 0, 0)
    End Sub
End Class
